### PR TITLE
Use new FWUP module

### PR DESCRIPTION
**Pull request by ConnorRigby**
This allows a few new things:
* Error handling from FWUP
* Progress reporting to NervesHub (if we so choose to do so)
* Client application interception

### CODE REVIEW

File: lib/nerves_hub/client/default.ex [hunk 1]
```diff
@@ -5,7 +5,25 @@ defmodule NervesHub.Client.Default do
...
 
   @behaviour NervesHub.Client
+  require Logger
 
   @impl NervesHub.Client
   def update_available(_), do: :apply
+
+  @impl NervesHub.Client
+  def handle_fwup_message({:progress, progress}) do
+    IO.write("FWUP PROG: #{progress}%\r")
```
[the remark this answers is not in the document]
Comment (ConnorRigby) [reply lib/nerves_hub/client/default.ex]: Yeah I think that is a good idea. :+1:

**Comment by ConnorRigby** [comment]
Yeah I hate the name but I couldn't think of anything either :/ I didn't want that to halt progress. I'd be happy to take name suggestions

**Comment by fhunleth** [comment]
:100: Let's not stall progress on the library name.

**Comment by mobileoverlord** [comment]
what about `libfwup`. Anything by `exfwup`. I think the first rule of elixir is to not name things `ex`-something

**Comment by fhunleth** [comment]
How about just `fwup`?

**Comment by ConnorRigby** [comment]
I like just `fwup`. :+1:

**Comment by ConnorRigby** [comment]
Naming problem taken care of. [here](https://github.com/ConnorRigby/elixir-fwup) is the repo

**Pull request merged**